leaf/extensions/tags.py

```diff
@@ -65,6 +66,28 @@ async def tag_autocomplete(
             for tag in tag_records
         ]
 
+    def is_tag_being_made(self, guild_id, name):
+        try:
+            being_made = self._reserved_tags_being_made[guild_id]
+        except KeyError:
+            return False
+        else:
+            return name.lower() in being_made
+
+    def add_in_progress_tag(self, guild_id, name):
+        tags = self._reserved_tags_being_made.setdefault(guild_id, set())
+        tags.add(name.lower())
+
+    def remove_in_progress_tag(self, guild_id, name):
+        try:
+            being_made = self._reserved_tags_being_made[guild_id]
+        except KeyError:
+            return
+
+        being_made.discard(name.lower())
+        if len(being_made) == 0:
```

Comment:
I'm curious if you could use `if not being_made` here? I do not work with sets very often

Reply:
Yup thats fine too. Added that to the latest commit.